From aeb795ef2290af1a0e4b14909363bc574e6b3ee7 Mon Sep 17 00:00:00 2001
From: Jack Yu <jackyu@google.com>
Date: Mon, 16 Jan 2017 10:15:34 -0800
Subject: Added permission check for setCellInfoListRate

Now the caller needs to have MODIFY_PHONE_STATE to call
this API.

Test: Manual
bug: 33815946
Merged-In: Ia2cf0de383572033c70b94ae71080ef35c3f443c
Change-Id: Ia2cf0de383572033c70b94ae71080ef35c3f443c
---
 src/com/android/phone/PhoneInterfaceManager.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index 4fc6ada..50f90c1 100644
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -1640,6 +1640,7 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
 
     @Override
     public void setCellInfoListRate(int rateInMillis) {
+        enforceModifyPermission();
         mPhone.setCellInfoListRate(rateInMillis);
     }
 
-- 
cgit v1.1

